### PR TITLE
#370: Strange Playlist and Song modal behaviour when clicking right button twice

### LINKED ISSUE #370: Strange Playlist and Song modal behaviour when clicking right button twice
**Describe the bug**

When opening a second Playlist or Song modal when another is opened we can see the first modal briefly before disappearing.

**To Reproduce**

1. Open a song or playlist modal. (Right click)
2. Open another modal across the page

https://github.com/user-attachments/assets/32448c63-184e-44a4-a3f6-498225a56ce3

**Expected behavior**

- When opening the second modal the first one should not appear for a moment
- When clicking  a second modal the first one should disappear and the second should pop up 
- See Spotify modals behavior when right clicking multiple songs or playlists 

**Screenshots**

**Additional context**

- See Spotify Song and Playlist modal behavior

---

__🖐️ Learn [how to CONTRIBUTE](https://antoniomrtz.github.io/SpotifyElectron_Web/docs/CONTRIBUTING/), all help is welcomed :)__

__📚 Check out the [project documentation](https://antoniomrtz.github.io/SpotifyElectron_Web/docs/) for more info__

__⭐ Help this project grow by giving it a star if you like it!__

__😵 Trello links are used for traceability, do not ask to join the board before leaving a comment in the related issue__

### PR DESCRIPTION
**Pull request by yajathb**
## Description

This pull request resolves Issue #370 where right-clicking to open a context menu while one was already open caused the menu to teleport and immediately close, requiring a second click. It also fixes a routing typo preventing the "Edit details" playlist modal button from working properly.

## Checklist

- [x] I have read and followed the [Contributing Guidelines](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/CONTRIBUTING.md).
- [x] I have followed the project's [git convention](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/developer/Git-Convention.md) for the Pull Request's name and git commits.
- [x] I have tested the changes and verified they work as intended. Automated tests (unit, integration, e2e as applicable) and manual tests (UI interactions, edge cases, error states, etc.).
- [x] I have set the pull request base to the appropriate `development-vx.xx` branch corresponding to the issue milestone.

## Commit type

- `fix`: used when fixing a bug or error in the code.

## Issue

#370

## Proposed Changes

- Added `BackdropProps` to all Context Menu `Popover` components across the app to safely intercept and stop `onContextMenu` propagation natively, closing the open menu.
- Applied `onClick` and `onContextMenu` `{e.stopPropagation()}` interceptors effectively bypassing the intrusive MUI backdrop masking logic.
- Disabled `transitionDuration` within `Popover` config across all context menus to correct the visual teleporting illusion bug.
- Fixed the routing typo in `ContextMenuPlaylist` where the "Edit details" button pointed to the non-existent `/playlists/` route instead of `/playlist/`, restoring its functionality.

## Potential Impact

- Context menus currently open across the app (in `Song`, `PlaylistCard`, `Sidebar`, etc.) will cleanly dismiss properly when attempting to right-click open another one.
- Right-clicking empty space will now properly trigger global window termination resolving the modal overlay natively.
- No direct negative regressions expected; all 70 frontend Jest tests successfully pass.

## Screenshots

N/A

## Assigned

@yajathb

### CODE REVIEW

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
I should be able to open with right click another modal if one is already open. 
With this implementation you cannot do it

**Comment by yajathb** [comment]
okay, will update pr asap

File: Electron/src/main/main.ts [hunk 1]
```diff
@@ -31,11 +31,11 @@ ipcMain.handle('copy-to-clipboard', async (event, dataToClipboard) => {
 });
 
 ipcMain.handle('load-previous-url', async () => {
-  mainWindow?.webContents.goBack();
+  mainWindow?.webContents.navigationHistory.goBack();
 });
 
 ipcMain.handle('load-forward-url', async () => {
-  mainWindow?.webContents.goForward();
+  mainWindow?.webContents.navigationHistory.goBack();
```
Comment (AntonioMrtz) [inline Electron/src/main/main.ts]: Why is this needed?

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

**Comment by AntonioMrtz** [comment]
Would you mind uploading a video on how the interactions will the modal work after your update? It will be really useful @yajathb

**Comment by yajathb** [comment]
will do

**Comment by yajathb** [comment]
[cinnamon-2026-04-11T155919-0700.webm](https://github.com/user-attachments/assets/2a955b45-5bc0-490f-aa86-a261648efedc)

the recording is attached. at the beginning, all inputs are right clicks to show that the box isn't being teleported. i tried right clicking on something that didnt have a menu, and it didnt opne, showing it works.

the sections regarding edit details is to show that a new box is being spawned with the correct right click action, not just teleporting the same box around.

if you need any more info please don't hesitate to ask

**Comment by yajathb** [comment]
hey i noticed my pr hasnt been reviewed yet, i just wanted to ask what the status is so far

**Comment by AntonioMrtz** [comment]
Hi @yajathb I will review it ASAP

File: Electron/src/components/Cards/SongCard/SongCard.tsx [hunk 2]
```diff
@@ -90,6 +91,15 @@ export default function SongCard({
...
         <Popover
+          key={
+            anchorPosition
+              ? `${anchorPosition.top}-${anchorPosition.left}`
+              : 'popover'
+          }
+          onClick={(e) => e.stopPropagation()}
+          onContextMenu={(e) => e.stopPropagation()}
+          hideBackdrop
+          disableScrollLock
```
Comment (AntonioMrtz) [inline Electron/src/components/Cards/SongCard/SongCard.tsx]: Why do we need this? In regular Spotify scroll is locked when context menu is open

File: README.md [hunk 3]
```diff
@@ -236,5 +236,5 @@ We're actively looking for new contributors 🙌. Feel free to check our [CONTRI
          <td></td>
          <td></td>
     </tr>
- 
+
```
Comment (AntonioMrtz) [inline README.md]: Can you add yourself in this file and the contributors one?

File: Electron/src/pages/Playlist/Playlist.tsx [hunk 4]
```diff
@@ -548,6 +549,15 @@ export default function Playlist({ refreshSidebarData }: PropsPlaylist) {
...
       <div>
         <Popover
+          key={
+            anchorPosition
+              ? `${anchorPosition.top}-${anchorPosition.left}`
+              : 'popover'
+          }
+          onClick={(e) => e.stopPropagation()}
+          onContextMenu={(e) => e.stopPropagation()}
+          hideBackdrop
```
Comment (AntonioMrtz) [inline Electron/src/pages/Playlist/Playlist.tsx]: Does this change something? I'm not seeing the backdrop color change when this is removed

File: Electron/src/main/main.ts (hunk 1 above)
Comment (AntonioMrtz) [reply Electron/src/main/main.ts]: Ok, this is because previous version was deprecated. Nice!

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

**Comment by AntonioMrtz** [comment]
Can you also check eslint pipeline failure?

**Comment by yajathb** [comment]
yeah will do, working on this right now

File: Electron/src/components/Cards/SongCard/SongCard.tsx (hunk 2 above)
Comment (yajathb) [reply Electron/src/components/Cards/SongCard/SongCard.tsx]: oh my bad i didnt know this; i use apple music

File: Electron/src/components/Cards/SongCard/SongCard.tsx [hunk 5]
```diff
@@ -11,7 +12,7 @@ export default function SongCard({
   name,
   artist,
   photo,
-  refreshSidebarData,
+  // refreshSidebarData,
```
Comment (AntonioMrtz) [inline Electron/src/components/Cards/SongCard/SongCard.tsx]: Simple remove it if not needed

File: README.md [hunk 6]
```diff
@@ -233,8 +233,13 @@ We're actively looking for new contributors 🙌. Feel free to check our [CONTRI
...
-         <td></td>
-         <td></td>
+        <td align="center">
+             <a href="https://github.com/yajathb">
+                 <img src="https://avatars.githubusercontent.com/u/66492618?v=4" width="90" alt="yajathb" />
+                 <br />
+                 <sub><b>yajathb</b></sub>
+             </a>
+        </td>
+        <td></td>
```
Comment (AntonioMrtz) [inline README.md]: I think this is a typo. Check README also

File: Electron/package-lock.json
Comment (AntonioMrtz) [inline Electron/package-lock.json:1]: Pull from main and run npm i. I don't think this should change package lock

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

File: Electron/package-lock.json
Comment (yajathb) [reply Electron/package-lock.json:1]: i ran npm audit because it was giving warnings about deprecated packages

Comment (AntonioMrtz) [reply Electron/package-lock.json:1]: Revert them. With main branch changes the should be fixed

**Comment by AntonioMrtz** [comment]
npm run format still failing

File: Electron/package-lock.json
Comment (AntonioMrtz) [inline Electron/package-lock.json:1]: Remove changes to this file

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

File: Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx [hunk 7]
```diff
@@ -71,6 +72,15 @@ export default function PlaylistSidebar({
       </div>
       <div>
         <Popover
+          key={
+            anchorPosition
+              ? `${anchorPosition.top}-${anchorPosition.left}`
+              : 'popover'
+          }
+          onClick={(e) => e.stopPropagation()}
+          onContextMenu={(e) => e.stopPropagation()}
```
Comment (AntonioMrtz) [inline Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx]: I don't think onclick and context menu stop prop is needed

File: Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx [hunk 8]
```diff
@@ -85,21 +95,30 @@ export default function PlaylistSidebar({
...
           sx={{
+            pointerEvents: 'none',
             '& .MuiPaper-root': {
               backgroundColor: 'var(--hover-white)',
+              pointerEvents: 'auto',
             },
-            '& . MuiPopover-root': {
+            '& .MuiPopover-root': {
               zIndex: '1000',
             },
```
[the remark this answers is not in the document]
Comment (AntonioMrtz) [reply Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx]: I think we can remove this safely

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

File: Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx [hunk 9]
```diff
@@ -85,21 +95,30 @@ export default function PlaylistSidebar({
...
-            handleCloseParent={handleClose}
-            owner={owner}
-            refreshPlaylistData={() => {}}
-            refreshSidebarData={refreshSidebarData}
-          />
+          <ClickAwayListener
+            onClickAway={handleClose}
+            mouseEvent="onPointerDown"
+          >
+            <div>
```
Comment (AntonioMrtz) [inline Electron/src/components/Sidebar/Playlist/PlaylistSidebar.tsx]: Is this div needed?
